### PR TITLE
Add README for sgcollect_info

### PR DESCRIPTION
**Pull request by tleyden**
Related to https://github.com/couchbase/sync_gateway/issues/3158

This doc should make it easier to use sgcollect_info and interpret results.  It also includes instructions on how to render raw profiles.

It's written in asciidoc, which is very similar to markdown but:

- Automatically generates a TOC
- Has more readable tables

I'm planning to give an asciidoc presentation to the team at some point.

### CODE REVIEW

**Comment by tleyden** [comment]
I forgot to mention that github automatically renders asciidoc.  Here is what the rendered file will look like when viewed on github:

https://github.com/couchbase/sync_gateway/blob/22a2ca08ea4320a9f12ad0308ab54d04e1701dcf/tools/README.adoc

File: tools/README.adoc [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+:toc: left
+:toclevels: 3
+
+
+== Overview
+
+This folder contains the following tools:
+
+. `sgcollect_info` -- a tool that collects diagnostics from Sync Gateway, such as logs, heap profile, cpu profile, expvars, etc.
+. A copy of `cbcollect_info` and `tasks.py`, which was refactored from the upstream `cbollect_info` in the nsserver repository, which has since diverged from this version.  `sgcollect_info` depends on the `cbcollect_info` in this repo.
```
Comment (JFlath) [inline tools/README.adoc]: (Nitpick) Should be `ns_server` not `nsserver`

File: tools/README.adoc [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+
+== SG Collectinfo
+
+=== Install dependencies (optional)
+
+To improve the quality of the collected zip files, install the following dependencies:
+
+- Go -- ideally use the version of Go that was used to build Sync Gateway
```
Comment (JFlath) [inline tools/README.adoc:20]: Would be awesome if we could drop in a table here over SG version -> Go version

File: tools/README.adoc [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+
+|===
+
+
+=== How to render raw heap files locally
+
+If `go` and/or `graphviz` is not installed on the machine where `sgcollect_info`, you can still render the raw profiles into a text or pdf formats with:
+
+. Modify `pprof_http_heap.log` to remove the "header" lines mentioned by @malarky in https://github.com/couchbase/sync_gateway/issues/3158#issuecomment-353565630 to avoid the error `pprof_http_heap.log: parsing profile: unrecognized profile format`
```
Comment (JFlath) [inline tools/README.adoc]: I'd prefer if we just said "the top 4 lines" and have the comment reference as an afterthought (i.e. I should know what I need to know from the readme, but can check the comment if I'm the curious sort).

**Review by JFlath — CHANGES_REQUESTED** [review]

File: tools/README.adoc (hunk 1 above)
Comment (tleyden) [reply tools/README.adoc]: I linked to the file on the nsserver repo

File: tools/README.adoc (hunk 2 above)
Comment (tleyden) [reply tools/README.adoc:20]: Dropped a table in, but I can't figure out the SG 1.5 version of Go that was used to build it.

@ceejatec / @kimtrang do you know how to find out which version of Go was used to build SG 1.5?

File: tools/README.adoc (hunk 3 above)
Comment (tleyden) [reply tools/README.adoc]: Yeah, good point

Comment (tleyden) [reply tools/README.adoc]: Fixed

File: tools/README.adoc [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+:toc: left
+:toclevels: 3
+
+
+== Overview
+
+This folder contains the following tools:
+
+. `sgcollect_info` -- a tool that collects diagnostics from Sync Gateway, such as logs, heap profile, cpu profile, expvars, etc.
+. A copy of `cbcollect_info` and `tasks.py`, which was refactored from the upstream `cbollect_info` in the [ns_server repository](https://github.com/couchbase/ns_server/blob/master/cbcollect_info), which has since diverged from this version.  `sgcollect_info` depends on the `cbcollect_info` in this repo.
```
Comment (bbrks) [inline tools/README.adoc]: Looks like AsciiDoc links should be in the form of `URL[link text]` to render properly.

File: tools/README.adoc (hunk 2 above)
Comment (bbrks) [reply tools/README.adoc:20]: @tleyden you can use the delve debugger to evaluate `runtime.buildVersion`. Here's an example using SG 1.5.1 CE.

```
✔ ~/Downloads/couchbase-sync-gateway/bin
16:58 $ dlv exec sync_gateway
Type 'help' for list of commands.
(dlv) b main.main
Breakpoint 1 set at 0x20b3 for main.main() /Users/jenkins/jenkins/workspace/sgw-unix-build@2/1.5.1/community/godeps/src/github.com/couchbase/sync_gateway/main.go:28
(dlv) c
> main.main() /Users/jenkins/jenkins/workspace/sgw-unix-build@2/1.5.1/community/godeps/src/github.com/couchbase/sync_gateway/main.go:28 (hits goroutine(1):1 total:1) (PC: 0x20b3)
(dlv) p runtime.buildVersion
"go1.7.4"
```

**Review by bbrks — CHANGES_REQUESTED** [review]
Looks good. I followed the instructions it all worked nicely.

I left a comment about link formatting, but it's only a minor issue.

File: tools/README.adoc (hunk 2 above)
Comment (tleyden) [reply tools/README.adoc:20]: Neat!

File: tools/README.adoc (hunk 4 above)
Comment (tleyden) [reply tools/README.adoc]: Old habits die hard ;-)

**Comment by tleyden** [comment]
@bbrks changes pushed, back on your plate

**Review by bbrks — APPROVED** [review]

**Review by JFlath — APPROVED** [review]

File: tools/README.adoc [hunk 5]
```diff
@@ -126,8 +126,8 @@ There are also some extra flags you can pass, which you can see by running `./sg
 
 === How to render raw heap files locally
 
-If `go` and/or `graphviz` is not installed on the machine where `sgcollect_info`, you can still render the raw profiles into a text or pdf formats with:
+If `go` and/or `graphviz` is not installed on the machine where `sgcollect_info` is run, you can still render the raw profiles into a text or pdf formats with:
 
-. Modify `pprof_http_heap.log` to remove the top for header lines to avoid the error `pprof_http_heap.log: parsing profile: unrecognized profile format`
-. Pass the path to the **actual sync gateway binary** that was collected in the sgcollect_info zipfile.  Eg: `go tool pprof -top /path/to/sync_gateway modified_pprof_http_heap.log`
+. Modify `pprof_http_heap.log` to remove the top four header lines to avoid the error `pprof_http_heap.log: parsing profile: unrecognized profile format`
```
Comment (tleyden) [inline tools/README.adoc]: yikes

**Pull request merged**